File: aula1/server/src/main/java/br/edu/utfpr/pb/pw25s/server/service/ProductService.java
package br.edu.utfpr.pb.pw25s.server.service;

import br.edu.utfpr.pb.pw25s.server.model.Product;

import java.util.List;

public interface ProductService {

    Product create(Product product);

    Product update(Product product);

    List<Product> findAll();

    Product findOne(Long id);

    void delete(Long id);

}

File: aula1/client/src/service/ProductService.ts
import { IProduct } from "@/commons/interfaces";
import { api } from "@/lib/axios";

const findAll = () => {
  return api.get("/products");
};

const save = (product: IProduct) => {
  return api.post("/products", product);
};

const update = (product: IProduct) => {
  return api.put("/products", product);
};

const remove = (id: number) => {
  return api.delete(`/products/${id}`);
};

const findOne = (id: number) => {
  return api.get(`/products/${id}`);
};

const ProductService = {
  findAll,
  save,
  remove,
  findOne,
};

export default ProductService;
